### PR TITLE
Ipmitool bookworm: Fix and patch enterprise-numbers URL

### PR DESCRIPTION
**Pull request by rajib-dutta1**
[ipmitool_1.8.19-4_amd64.deb.log](https://github.com/sonic-net/sonic-buildimage/files/14029877/ipmitool_1.8.19-4_amd64.deb.log)

#### Why I did it

ipmitool utility is used to access various HW sensors. Some platforms use "ipmitool raw " to read specific addresses. 

ipmitool_1.8.19-4_amd64.deb, that is part of bookworm has a defect. The package is missing file enterprise.txt that is expected by the "raw read" code path. 
It is so because the file the .deb tries to download at the build time does not have the necessary extension as it is available on remote server: https://www.iana.org/assignments/enterprise-numbers.txt

##### Work item tracking
NA

#### How I did it

The defect had been fixed using coding changes in next unstable version of Linux. It is expected to be available in future stable version of the OS. Hence to keep the changes to minimal, the .dsc file is downloaded and only the Makefile is modified to download the correct file. To make is work as patch necessary changes are made.

#### How to verify it
Build log is attached and installation of the file is noted line #2274
When using vanilla bookworm on platforms like 5212 or 5224:
-------------------------------------------------------------------
root@sonic:~# ipmitool raw 0x04 0x2d 0x31
IANA PEN registry open failed: No such file or directory
00 c0 01 80

When fixed we should not see the above error:
--------------------------------------------------
root@sonic:/home/admin# ipmitool raw 0x04 0x2d 0x31
 00 c0 00 80

#### Which release branch to backport (provide reason below if selected)
None
 
#### Tested branch (Please provide the tested image version)

bookworm

#### Description for the changelog

This change is to address ipmitool raw read issue. This patch must be removed once it is available in next stable Linux release that contains the fix. 

https://github.com/ipmitool/ipmitool/commit/1edb0e27e44196d1ebe449aba0b9be22d376bcb6

PR#17878

#### Link to config_db schema for YANG module changes
NA

### CODE REVIEW

File: files/build_templates/sonic_debian_extension.j2 [hunk 1]
```diff
@@ -101,6 +101,10 @@ sudo mkdir -p $FILESYSTEM_ROOT_USR_LIB_SYSTEMD_SYSTEM
 sudo dpkg --root=$FILESYSTEM_ROOT -i $debs_path/ifupdown2_*.deb || \
     sudo LANG=C DEBIAN_FRONTEND=noninteractive chroot $FILESYSTEM_ROOT apt-get -y install -f
 
+# Install a patched version of ipmitool  (and its dependencies via 'apt-get -y install -f')
+sudo dpkg --root=$FILESYSTEM_ROOT -i $debs_path/ipmitool_*.deb || \
+    sudo LANG=C DEBIAN_FRONTEND=noninteractive chroot $FILESYSTEM_ROOT apt-get -y install -f
```
Comment (saiarcot895) [inline files/build_templates/sonic_debian_extension.j2:106]: In `build_debian.sh`, can you remove the `ipmitool` package installation from there, to make sure this version is the only version that gets installed?

Comment (rajib-dutta1) [reply files/build_templates/sonic_debian_extension.j2:106]: I pushed the changed asked above. 

root@sonic:/home/admin# show ver

SONiC Software Version: SONiC.master.0-dirty-20240209.233855
SONiC OS Version: 12
Distribution: Debian 12.4
Kernel: 6.1.0-11-2-amd64
Build commit: bd47fd155
Build date: Sat Feb 10 01:35:50 UTC 2024
Built by: commbuild@rajvm

Platform: x86_64-dellemc_s5212f_c3538-r0
HwSKU: DellEMC-S5212f-P-25G
ASIC: broadcom
ASIC Count: 1
Serial Number: CQY67C3
Model Number: 0VK93C
Hardware Revision: 01
Uptime: 07:59:58 up 4 min,  1 user,  load average: 5.00, 4.12, 1.78
Date: Tue 13 Feb 2024 07:59:58


root@sonic:/home/admin# ipmitool raw 0x04 0x2d 0x31
Get Device ID command failed: 0xb1 Unknown (0xB1)
Received a response with unexpected ID 0 vs. 1
 00 c0 01 80


root@sonic:/home/admin# dpkg -l ipmitool
||/ Name           Version      Architecture Description
+++-==============-============-============-==================================>
ii  ipmitool       1.8.19-4     amd64        utility for IPMI control with kern>

**Comment by saiarcot895** [comment]
Can you update the title to show what is being fixed?

**Comment by rajib-dutta1** [comment]
> Can you update the title to show what is being fixed?

Done.

**Review by saiarcot895 — APPROVED** [review]

**Review by aravindmani-1 — APPROVED** [review]

**Comment by jeff-yin** [comment]
@saiarcot895 https://github.com/sonic-net/sonic-buildimage/pull/18094 is merged and it looks like the pipelines have passed. Can this be merged?

**Comment by jeff-yin** [comment]
@qiluo-msft @lguohan @xumia please review/approve/merge, thanks. This is needed as part of the bookworm upgrade.

**Review by qiluo-msft — APPROVED** [review]

**Pull request merged**